src/test/credentials/provider/ec2CredentialsProvider.test.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { AWSError, EC2MetadataCredentials, MetadataService } from 'aws-sdk'
+import { anyFunction, anyString } from 'ts-mockito'
+import { Ec2CredentialsProvider } from '../../../credentials/providers/ec2CredentialsProvider'
+import { instance, mock, when } from '../../utilities/mockito'
+
+describe('Ec2CredentialsProvider', () => {
+    const dummyRegion = 'dummmyRegion'
+
+    const mockResponse = JSON.stringify({
+        region: dummyRegion,
+    })
+
+    let mockMetadata: MetadataService
+    let credentialsProvider: Ec2CredentialsProvider
+
+    beforeEach(function () {
+        mockMetadata = mock()
+        credentialsProvider = new Ec2CredentialsProvider(instance(mockMetadata))
+    })
+
+    afterEach(() => {})
+
+    it('should be valid if EC2 metadata service resolves', async () => {
```

Comment:
Nit (throughout): use `function() {` instead of `() => {` for Mocha tests: https://mochajs.org/#arrow-functions